plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/AbstractXmlSourceValidator.java

```diff
@@ -79,6 +96,38 @@ void createMessage(IReporter reporter, BannedElement element, int elementOffset,
     reporter.addMessage(this, message);
   }
   
+  /**
+   * Returns the underlying IProject from a given IValidationContext or
+   * null if the IValidationContext does not return any files that need
+   * to be validated.
+   */
+  IProject getProject(IValidationContext helper) {
+    String[] fileUri = helper.getURIs();
+    if (fileUri.length > 0) {
+      IFile file = getFile(fileUri[0]);
+      if (file != null) {
+        return file.getProject();
+      }
+    }
+    return null;
+  }
+  
+  /**
+   * Returns the IFile for a given URI or null if the file does
+   * not exist in the workspace.
+   */
+  IFile getFile(String fileUri) {
+    IPath path = new Path(fileUri);
```

Comment:
Yep, IncrementalHelper.setURI() takes in a string, and I was using file.getLocationURI().toString(), which began with file:, when I actually needed the path.